packages/reactivity/src/ref.ts

```diff
@@ -85,6 +85,11 @@ function toProxyRef<T extends object, K extends keyof T>(
 
 type UnwrapArray<T> = { [P in keyof T]: UnwrapRef<T[P]> }
 
+// corner case when use narrows type
+// Ex. type RelativePath = string & { __brand: unknown }
+// RelativePath extends object -> true
+type BaseTypes = string | number
```

Comment:
should this include `boolean` too?

Reply:
I think `boolean` not need special type.
but also can add it.